plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.test.ts

```diff
@@ -381,6 +410,103 @@ describe('DatabaseEntitiesCatalog', () => {
       expect(result).toEqual([{ entityId: 'u' }]);
     });
 
+    it('should update if attachments are changed', async () => {
+      const entity: Entity = {
+        apiVersion: 'a',
+        kind: 'b',
+        metadata: {
+          uid: 'u',
+          etag: 'e',
+          generation: 1,
+          name: 'c',
+          namespace: 'd',
+        },
+        spec: {
+          x: 'a',
+        },
+      };
+
+      db.entities.mockResolvedValue({
+        entities: [{ entity }],
+        pageInfo: { hasNextPage: false },
+      });
+      db.entityByUid.mockResolvedValue({ entity });
+      db.updateEntity.mockResolvedValue({ entity });
+      db.attachmentsByUid.mockResolvedValue([
+        {
+          key: 'a',
+          entityUid: 'u',
+          contentType: 'text/plain',
+          etag: 'tttt',
+        },
+      ]);
+
+      const catalog = new DatabaseEntitiesCatalog(db, getVoidLogger());
+      const result = await catalog.batchAddOrUpdateEntities([
+        {
+          entity,
+          relations: [],
+          attachments: [
+            {
+              key: 'a',
+              content: {
+                contentType: 'text/plain',
+                data: Buffer.from('Hello World'),
+              },
+            },
+          ],
+        },
+      ]);
+
+      expect(db.attachmentsByUid).toHaveBeenCalledTimes(1);
+      expect(db.attachmentsByUid).toHaveBeenCalledWith(expect.anything(), 'u');
+      expect(db.entities).toHaveBeenCalledTimes(1);
+      expect(db.entities).toHaveBeenCalledWith(expect.anything(), {
+        filter: basicEntityFilter({
+          kind: 'b',
+          'metadata.namespace': 'd',
+          'metadata.name': 'c',
+        }),
+      });
+      expect(db.entityByName).not.toHaveBeenCalled();
+      expect(db.entityByUid).toHaveBeenCalledTimes(1);
+      expect(db.entityByUid).toHaveBeenCalledWith(transaction, 'u');
+      expect(db.updateEntity).toHaveBeenCalledTimes(1);
+      expect(db.updateEntity).toHaveBeenCalledWith(
+        transaction,
+        {
+          entity: {
+            apiVersion: 'a',
+            kind: 'b',
+            metadata: {
+              uid: 'u',
+              etag: expect.any(String),
+              // TODO: Do we want to increase the entity generation if attachments change? What about the etag?
+              generation: 1,
```

Comment:
Hmm, I'm not sure about this, what is your opinion? @freben @Rugvip

Reply:
Generation is likely going away, but the answer would be no since it's only based on `spec`. For `etag` I would say yes, as it's used to inform whether you need to update your local representation of the entity, e.g. attachments added

Reply:
Just thinking about this. Right now it's using `entityHasChanges` to detect whether an entity has changed and whether to increase generation and to update the etag. `entityHasChanges` takes two entities and compares them. `entityHasChanges` comes from `catalog-model` and till now the `Entity` doesn't contain any details about the attachments. It's a completely separate concept.

I think we have two options:
* Either we change how `entityHasChanges` detect changes or have separate code that regenerates the etag if the attachments have changes in `DatabaseEntitiesCatalog`.
* We extend the `Entity` to include details about available attachments. Right now, we don't expose which attachments are available for an entity. This is completely hidden and consumers have to know which keys they are looking for. This could actually help with discover of this feature. It could look like:
    ```yaml
    apiVersion: backstage.io/v1alpha1
    kind: API
    metadata:
      name: petstore
    spec:
      type: openapi
      lifecycle: experimental
      owner: team-c
      definitionLocation: attachment:backstage.io/api-definition
    attachments:
      - key: backstage.io/api-definition
        contentType: text/plain
        etag: sdasdasd
    ```

Any thoughts?

Reply:
For the second option we would have to make sure that we don't store the attachments property in the database but generate it on read from the attachments table. However, as said before, it's not desired to add more top level stuff to `Entity`. On the other side, it's exactly like `relations` are done.

Reply:
On the other hand, emitting different relations doesn't trigger a change to the etag now either. Hmm, not sure how to continue here.

Reply:
So in catalog v2 we stick together final entities whenever something changes about an entity. Essentially we read from a couple of different tables, e.g. the main processed entities table and the relations table, and put that all into the final entity. That's the point where we generate the etag as well, so we'll be able to include relations in the etag now.

So once we're on v2 it'll be pretty simple to implement this, the stitcher would simply look up all attachments for the the entity in the attachments table and write it to the attachments array, if that's how we end up modeling them.

Reply:
So I guess it would be easier to skip that for now at include it in the v2 rework - the impact of not updating the etag is small, as the caching is done on the attachment itself. So consumers get the latest content, even though we didn't updated the etag.